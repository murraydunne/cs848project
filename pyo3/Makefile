.PHONY: test test_py3 publish clippy lint fmt

# Constants used in clippy target
CLIPPY_LINTS_TO_DENY := warnings
CLIPPY_LINTS_TO_ALLOW := clippy::new_ret_no_self

test:
	cargo test
	${MAKE} clippy
	tox
	for example in examples/*; do tox -e py -c $$example/tox.ini; done

test_py3:
	tox -e py3
	for example in examples/*; do tox -e py3 -c $$example/tox.ini; done

fmt:
	cargo fmt --all -- --check
	black . --check

clippy:
	@touch src/lib.rs  # Touching file to ensure that cargo clippy will re-check the project
	cargo clippy --all-features --all-targets -- \
		$(addprefix -D ,${CLIPPY_LINTS_TO_DENY}) \
		$(addprefix -A ,${CLIPPY_LINTS_TO_ALLOW})

lint: fmt clippy
	@true

publish: test
	cargo publish --manifest-path pyo3-derive-backend/Cargo.toml
	cargo publish --manifest-path pyo3cls/Cargo.toml
	cargo publish
