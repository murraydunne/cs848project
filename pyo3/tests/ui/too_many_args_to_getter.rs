use pyo3::prelude::*;

#[pyclass]
struct ClassWithGetter {
    a: u32,
}

#[pymethods]
impl ClassWithGetter {
    #[getter]
    fn get_num(&self, index: u32) {}
}

fn main() {}
