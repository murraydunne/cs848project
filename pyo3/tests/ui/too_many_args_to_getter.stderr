error: Getter function can only have one argument of type pyo3::Python!
  --> $DIR/too_many_args_to_getter.rs:11:30
   |
11 |     fn get_num(&self, index: u32) {}
   |                              ^^^
